--- a/commands/Kconfig
+++ b/commands/Kconfig
@@ -2755,3 +2755,7 @@
 endmenu
 
 endif
+
+config CMD_ZQCAL
+	bool "K4 NT ROM RAM ZQ calibration measurement"
+	depends on MACH_KINDLE_MX50
--- a/arch/arm/boards/kindle-mx50/Makefile
+++ b/arch/arm/boards/kindle-mx50/Makefile
@@ -3,3 +3,5 @@
 obj-y += board.o
 lwl-y += lowlevel.o
 bbenv-y += defaultenv-kindle-mx50
+
+obj-$(CONFIG_CMD_ZQCAL) += zqcal.o zqcal-sram.o early-trace.o
